deepspeed/profiling/flops_profiler/README.md

```diff
@@ -35,216 +38,324 @@ Top 3 modules in flops at depth 2 are {'Conv2d': '421.91 MMACs', 'Linear': '11.1
+            )
+            ...
+            (3): ParallelTransformerLayer(...)
+          (final_layernorm): FusedLayerNorm(1.02 k, 0.00% Params, 0 MACs, 0.00% MACs, 52.69 us, 0.16% time, 0.0 TFLOPS, torch.Size([512]), eps=1e-05, elementwise_affine=True)
+        )
+      )
+    )
+  )
+)
+Top 3 modules in flops at depth 8 are {'ColumnParallelLinear': '60.13 GMACs', 'RowParallelLinear': '42.95 GMACs', 'FusedScaleMaskSoftmax': '536.87 MMACs'}
+Top 3 modules in params at depth 8 are {'ColumnParallelLinear': '7.35 M', 'RowParallelLinear': '5.25 M', 'FusedScaleMaskSoftmax': '0'}
+Top 3 modules in time at depth 8 are {'ColumnParallelLinear': '595.81 us', 'RowParallelLinear': '529.29 us', 'FusedScaleMaskSoftmax': '334.26 us'}
+Number of MACs:                 103616086016
+Number of parameters:           38890496
+```
 
-  # forward() method
-  loss = model(batch)
+##  Usage Outside the DeepSpeed Runtime
 
-  # runs backpropagation
-  loss.backward()
+The flops profiler can be used as a standalone package outside of the DeepSpeed runtime.
+One can simply install DeepSpeed and import the `flops_profiler` package to use the APIs directly.
+Refer to [installation of DeepSpeed](https://www.deepspeed.ai/getting-started/#installation) for installing DeepSpeed.
 
-  # weight update
-  optimizer.step()
-```
+### In Model Inference
 
-#### Use the high level-API and run the model inference for profiling purpose
+To profile a trained model in inference, use the `get_model_profile` function.
+Examples are given below.
 
-Examples of this usage are given below.
+#### Example: AlexNet
 
-##### Classification model example:
+The following example shows how to profile AlexNet using the DeepSpeed flops profiler.
 
 ```python
-import argparse
-import sys
-import torch
 import torchvision.models as models
+import torch
 from deepspeed.profiling.flops_profiler import get_model_profile
 
-pt_models = {
-    'resnet18': models.resnet18,
-    'resnet50': models.resnet50,
-    'alexnet': models.alexnet,
-    'vgg16': models.vgg16,
-    'squeezenet': models.squeezenet1_0,
-    'densenet': models.densenet161,
-    'inception': models.inception_v3
-}
-
-if __name__ == '__main__':
-    parser = argparse.ArgumentParser(description='flops-profiler example script')
-    parser.add_argument('--device',
-                        type=int,
-                        default=0,
-                        help='Device to store the model.')
-    parser.add_argument('--model',
-                        choices=list(pt_models.keys()),
-                        type=str,
-                        default='resnet18')
-    args = parser.parse_args()
-
-    model = pt_models[args.model]()
-
-    if torch.cuda.is_available():
-        model.cuda(device=args.device)
-
+with torch.cuda.device(0):
+    model = models.alexnet()
     batch_size = 256
-    macs, params, steps = get_model_profile(model, # the PyTorch model to be profiled
+    macs, params = get_model_profile(model=model, # model
```

Comment:
It would be good to how how the  model is called without the profiling so that the input_res and input_consturctors are clearer